rosbag2_compression/include/rosbag2_compression/compression_factory.hpp

```diff
@@ -53,7 +53,7 @@ class ROSBAG2_COMPRESSION_PUBLIC CompressionFactory
    * \return A unique pointer to the newly created compressor.
    * \throw invalid_argument If the compression format does not exist.
    */
-  std::unique_ptr<rosbag2_compression::BaseCompressorInterface>
+  virtual std::unique_ptr<rosbag2_compression::BaseCompressorInterface>
```

Comment:
What's the reason to make this and the one below virtual?

Reply:
See my PR description. It's so that the method could be mocked.